File: src/Test/HelloWorld.cs
namespace Test
{
    [TestClass]
    public class HelloWorld
    {
        [TestMethod]
        public void TestMethod()
        {
            using var bus = new MessageBus();
            var subscriberTicket = bus.RegisterSubscriber<string>("Hello", SubscriberMethod);
            var text = "Hello World";

            var publisherTicket = bus.RegisterPublisher<string>("Hello");
            publisherTicket.Executor.Execute(text);

            bus.UnregisterPublisher(publisherTicket);
            bus.UnregisterSubscriber(subscriberTicket);

            Assert.AreEqual(text, _received);
        }

        private string? _received;
        public void SubscriberMethod(string? argument)
        {
            _received = argument;
        }
    }
}

File: src/TestVB/HelloWorld.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports SecretNest.MessageBus

<TestClass>
Public Class HelloWorld
    <TestMethod>
    Sub TestMethod()
        Using bus As New MessageBus
            Dim subscriberTicket = bus.RegisterSubscriber(Of String)("Hello", AddressOf SubscriberMethod)
            Dim text = "Hello World"

            Dim publisherTicket = bus.RegisterPublisher(Of String)("Hello")
            publisherTicket.Executor.Execute(text)

            bus.UnregisterPublisher(publisherTicket)
            bus.UnregisterSubscriber(subscriberTicket)

            Assert.AreEqual(text, _received)
        End Using
    End Sub

    Private _received As String
    Public Sub SubscriberMethod(argument As String)
        _received = argument
    End Sub
End Class
